packages/ui/src/common/email-or-username-or-phone-number-field.tsx

```diff
@@ -10,20 +11,18 @@ import { PhoneNumberField } from './phone-number-field';
 export function EmailOrUsernameOrPhoneNumberField({
   className,
   name = undefined,
-  labelEmailOrUsername = 'Email address or username',
-  labelPhoneNumber = 'Phone number',
-  toggleLabelEmailOrUsername = 'Use email or username',
-  toggleLabelPhoneNumber = 'Use phone',
-  locationBasedCountryIso,
-  toggleDescription = 'Toggle between email or username, and phone.',
+  toggleLabelEmailOrUsername,
+  toggleLabelPhoneNumber,
   ...props
 }: {
-  labelEmailOrUsername?: React.ReactNode;
-  labelPhoneNumber?: React.ReactNode;
-  toggleLabelEmailOrUsername?: string;
-  toggleLabelPhoneNumber?: string;
-  locationBasedCountryIso: React.ComponentProps<typeof PhoneNumberField>['locationBasedCountryIso'];
-  toggleDescription?: string;
+  /**
+   * **Note:** this prop is required as the translation differs between `signIn` and `signUp`
+   */
+  toggleLabelEmailOrUsername: React.ReactNode;
```

Comment:
This use of inline docs came from a chat with @dstaley; making sure any unique/different approaches are accompanied by implementation details

Additionally, by using comment blocks, we can surface these notes through the props